docs/src/starknet/calldata-transformation.md

```diff
@@ -0,0 +1,206 @@
+
+### Basic example
+
+We can write the same command as above, but with expression calldata:
+
+```shell
+$ sncast --account myuser \
+    call \
+    --url http://127.0.0.1:5050 \
+    --contract-address 0x016ad425af4585102e139d4fb2c76ce786d1aaa1cfcd88a51f3ed66601b23cdd \
+    --function tuple_fn \
+    --calldata (0x10, 3, data_stransformer_contract::Enum::One) \
+    --block-id latest
+```
+
+getting the same result.
+
+> 📝 **Note**
+> All data types are serialized according to the official [Starknet specification](https://docs.starknet.io/architecture-and-concepts/smart-contracts/serialization-of-cairo-types/).
+
+> 📝 **Note**
+> User-defined items such as enums and structs should be referred to depending on a way they are defined in ABI.
+> In general, paths to items have form: `<project-name>::<module-path>::<item-name>`.
+
+## Supported Expressions
+
+> 💡 **Info**
+> Only **constant** expressions are supported. Defining and referencing variables and calling functions (either builtin, user-defined or external) is not allowed.
+
+Cast supports most important Cairo corelib types:
+  * `bool`
+  * signed integers (`i8`, `i16`, `i32`, `i64`, `i128`)
+  * unsigned integers (`u8`, `u16`, `u32`, `u64`, `u96`, `u128`, `u256`, `u384`, `u512`)
+  * `felt252` (numeric literals and so-called *shortstrings*)
+  * `ByteArray`
+  * `ContractAddress`
+  * `ClassHash`
+  * `StorageAddress`
+  * `EthAddress`
+  * `bytes31`
+  * `Array` - using `array![]` macro
+
+Numeric types (primitives and `felt252`) can be paseed with type suffix specified -\
+for example `--calldata 420_u64`.
+
+### More complex examples
+
+  1. `complex_fn` - different data types:
+
+  ```shell
+  $ sncast --account myuser \
+      call \
+      --url http://127.0.0.1:5050 \
+      --contract-address 0x016ad425af4585102e139d4fb2c76ce786d1aaa1cfcd88a51f3ed66601b23cdd \
+      --function complex_fn \
+      --calldata \
+          array![array![1, 2], array![3, 4, 5], array![6]], \
+          12, \
+          -128_i8, \
+          "Some string (a ByteArray)", \
+          ('a shortstring', 32_u32), \
+          true, \
+          0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff \
+      --block-id latest
+  ```
+
+  Note the commas separating the arguments.
+
+  2. `nested_struct_fn` - struct nesting:
+
+  ```shell
+  $ sncast --account myuser \
+      call \
+      --url http://127.0.0.1:5050 \
+      --contract-address 0x016ad425af4585102e139d4fb2c76ce786d1aaa1cfcd88a51f3ed66601b23cdd \
+      --function nested_struct_fn \
+      --calldata \
+          data_transformer_contract::NestedStructWithField { \
+              a: data_transformer_contract::SimpleStruct { a: 10 }, \
+              b: 12 \
+          } \
+      --block-id latest
+  ```
+
+## Caveats
+
+> 💡 **Info**
+> To force transformation of a single numeric literal, simply add a type suffix to it.
+
+> 💡 **Info**
+> To force transformation of a list of `felt252`, separate its elements with commas.
```

Comment:
I think these could be subsection in here, we have a bunch of info and similar in this page already